### PR TITLE
[XLA:CPU][oneDNN] Add a BF16 pattern for Softmax

### PR DESCRIPTION
**Pull request by akhilgoe**
This PR:

1. Adds a pattern for BF16 softmax.
2. Adds a test to verify rewrite and execution result.
3. Skips BF16 matmul / softmax tests on machines that do not support BF16

### CODE REVIEW

**Review by tdanyluk — APPROVED** [review]
Thanks for the contribution, looks fine to me.

File: xla/service/cpu/onednn_ops_rewriter.cc [hunk 1]
```diff
@@ -423,17 +429,12 @@ class OneDnnOpsRewriterVisitor : public DfsHloRewriteVisitor {
     return OkStatus();
   }
 
-  Status HandleConvert(HloInstruction* instr) override {
-    HloInstruction* ln_instr;
-    HloInstruction* convert_instr;
```
Comment (penpornk) [inline xla/service/cpu/onednn_ops_rewriter.cc:428]: `convert_instr` is still used in lines 442-443 (of the latest commit).

[XLA Linux CPU CI](https://source.cloud.google.com/results/invocations/f288bd54-030a-4d60-b0d4-dc90067c8f7d) error message:
```
external/xla/xla/service/cpu/onednn_ops_rewriter.cc:442:12: error: use of undeclared identifier 'convert_instr'
  442 |           (convert_instr->shape().element_type() == PrimitiveType::BF16) ||
      |            ^
external/xla/xla/service/cpu/onednn_ops_rewriter.cc:443:12: error: use of undeclared identifier 'convert_instr'
  443 |           (convert_instr->shape().element_type() == PrimitiveType::F16);
      |            ^
11 warnings and 2 errors generated.
```

**Review by penpornk — CHANGES_REQUESTED** [review]

File: xla/service/cpu/onednn_ops_rewriter.cc (hunk 1 above)
Comment (akhilgoe) [reply xla/service/cpu/onednn_ops_rewriter.cc:428]: Sorry about that...I missed it while merging main to resolve conflicts on my branch.

**Comment by akhilgoe** [comment]
@penpornk I have fixed the error due to merge. Can you please take a look whenever possible? Thanks!

**Review by penpornk — APPROVED** [review]
Thank you for the quick fix!

**Pull request closed without merging**